// materialize-angular/esm5/app/completed-components/button-toggle-group/button-toggle/button-toggle.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @license
 * Copyright Workylab. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://raw.githubusercontent.com/workylab/materialize-angular/master/LICENSE
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { config } from '../../../config';
var ButtonToggleComponent = /** @class */ (function () {
    function ButtonToggleComponent() {
        this.className = ButtonToggleComponent.defaultProps.className;
        this.disabled = ButtonToggleComponent.defaultProps.disabled;
        this.name = ButtonToggleComponent.defaultProps.name;
        this.value = ButtonToggleComponent.defaultProps.value;
        this.prefix = config.components.prefix;
        this.onClickEmitter = new EventEmitter();
        this.isActive = false;
        this.isFocused = false;
    }
    /**
     * @return {?}
     */
    ButtonToggleComponent.prototype.onClick = /**
     * @return {?}
     */
    function () {
        if (!this.disabled) {
            this.isActive = !this.isActive;
            this.isFocused = false;
            this.onClickEmitter.emit(this.value);
        }
    };
    /**
     * @return {?}
     */
    ButtonToggleComponent.prototype.onBlur = /**
     * @return {?}
     */
    function () {
        this.isFocused = false;
    };
    /**
     * @return {?}
     */
    ButtonToggleComponent.prototype.onFocus = /**
     * @return {?}
     */
    function () {
        if (!this.disabled) {
            this.isFocused = true;
        }
    };
    ButtonToggleComponent.defaultProps = {
        className: '',
        disabled: false,
        name: '',
        value: ''
    };
    ButtonToggleComponent.decorators = [
        { type: Component, args: [{
                    selector: config.components.prefix + "-button-toggle }",
                    template: "<div (click)=\"onClick()\" (blur)=\"onBlur()\" (focus)=\"onFocus()\"\n\n  [ngClass]=\"[prefix + '-button-toggle', className]\"\n  [class.active]=\"isActive\"\n  [class.disabled]=\"disabled\"\n  [class.focused]=\"isFocused\"\n  [tabindex]=\"disabled ? '-1': '0'\"\n  [isRippleActive]=\"!disabled\"\n\n  materializeRipple  \n>\n  <ng-content></ng-content>\n</div>\n"
                }] }
    ];
    /** @nocollapse */
    ButtonToggleComponent.ctorParameters = function () { return []; };
    ButtonToggleComponent.propDecorators = {
        onClickEmitter: [{ type: Output, args: ['onClick',] }],
        className: [{ type: Input }],
        disabled: [{ type: Input }],
        name: [{ type: Input }],
        value: [{ type: Input }]
    };
    return ButtonToggleComponent;
}());
export { ButtonToggleComponent };
if (false) {
    /** @type {?} */
    ButtonToggleComponent.defaultProps;
    /** @type {?} */
    ButtonToggleComponent.prototype.onClickEmitter;
    /** @type {?} */
    ButtonToggleComponent.prototype.className;
    /** @type {?} */
    ButtonToggleComponent.prototype.disabled;
    /** @type {?} */
    ButtonToggleComponent.prototype.name;
    /** @type {?} */
    ButtonToggleComponent.prototype.value;
    /** @type {?} */
    ButtonToggleComponent.prototype.prefix;
    /** @type {?} */
    ButtonToggleComponent.prototype.isActive;
    /** @type {?} */
    ButtonToggleComponent.prototype.isFocused;
}
