// materialize-angular/esm5/app/completed-components/slider/slider-option/slider-option.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @license
 * Copyright Workylab. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://raw.githubusercontent.com/workylab/materialize-angular/master/LICENSE
 */
import { Component, ElementRef, Input, ViewChild } from '@angular/core';
import { config } from '../../../config';
var SliderOptionComponent = /** @class */ (function () {
    function SliderOptionComponent() {
        this.className = SliderOptionComponent.defaultProps.className;
        this.value = SliderOptionComponent.defaultProps.value;
        this.prefix = config.components.prefix;
        this.isActive = false;
    }
    SliderOptionComponent.defaultProps = {
        className: '',
        value: null
    };
    SliderOptionComponent.decorators = [
        { type: Component, args: [{
                    selector: config.components.prefix + "-slider-option }",
                    template: "<div [ngClass]=\"[prefix + '-slider-option-container', className]\"\n  [class.active]=\"isActive\"\n\n  #template>\n  <ng-content></ng-content>\n</div>\n"
                }] }
    ];
    /** @nocollapse */
    SliderOptionComponent.ctorParameters = function () { return []; };
    SliderOptionComponent.propDecorators = {
        templateRef: [{ type: ViewChild, args: ['template', { static: true },] }],
        className: [{ type: Input }],
        value: [{ type: Input }]
    };
    return SliderOptionComponent;
}());
export { SliderOptionComponent };
if (false) {
    /** @type {?} */
    SliderOptionComponent.defaultProps;
    /** @type {?} */
    SliderOptionComponent.prototype.templateRef;
    /** @type {?} */
    SliderOptionComponent.prototype.className;
    /** @type {?} */
    SliderOptionComponent.prototype.value;
    /** @type {?} */
    SliderOptionComponent.prototype.prefix;
    /** @type {?} */
    SliderOptionComponent.prototype.isActive;
}
