// materialize-angular/esm2015/app/completed-components/drawer/drawer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @license
 * Copyright Workylab. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://raw.githubusercontent.com/workylab/materialize-angular/master/LICENSE
 */
import { Component, ElementRef, EventEmitter, Input, Output, Renderer2, ViewChild } from '@angular/core';
import { DRAWER_POSITIONS, DRAWER_TYPES } from './drawer.model';
import { config } from '../../config';
export class DrawerComponent {
    /**
     * @param {?} renderer
     */
    constructor(renderer) {
        this.renderer = renderer;
        this.className = DrawerComponent.defaultProps.className;
        this.closeOnBackdrop = DrawerComponent.defaultProps.closeOnBackdrop;
        this.hasBackdrop = DrawerComponent.defaultProps.hasBackdrop;
        this.isOpen = DrawerComponent.defaultProps.isOpen;
        this.position = DrawerComponent.defaultProps.position;
        this.transitionDuration = DrawerComponent.defaultProps.transitionDuration;
        this.type = DrawerComponent.defaultProps.type;
        this.prefix = config.components.prefix;
        this.close = this.close.bind(this);
        this.closeByBackdrop = this.closeByBackdrop.bind(this);
        this.open = this.open.bind(this);
        this.onOpenEmitter = new EventEmitter();
        this.onCloseEmitter = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        setTimeout((/**
         * @return {?}
         */
        () => {
            this.renderer.setStyle(this.drawerRef.nativeElement, 'transitionDuration', `${this.transitionDuration}ms`);
            if (this.closeOnBackdrop) {
                this.backdropRef.nativeElement.addEventListener('click', this.closeByBackdrop);
            }
        }), 0);
    }
    /**
     * @return {?}
     */
    open() {
        if (this.type !== DRAWER_TYPES.STATIC) {
            this.renderer.setStyle(this.backdropRef.nativeElement, 'display', 'block');
            setTimeout((/**
             * @return {?}
             */
            () => {
                this.isOpen = true;
            }), 0);
        }
    }
    /**
     * @param {?} event
     * @return {?}
     */
    closeByBackdrop(event) {
        if (event.target === this.backdropRef.nativeElement) {
            this.close();
        }
    }
    /**
     * @return {?}
     */
    close() {
        this.isOpen = false;
        setTimeout((/**
         * @return {?}
         */
        () => {
            this.renderer.setStyle(this.backdropRef.nativeElement, 'display', 'none');
            this.onCloseEmitter.emit();
        }), this.transitionDuration);
    }
}
DrawerComponent.defaultProps = {
    className: '',
    closeOnBackdrop: true,
    hasBackdrop: true,
    isOpen: false,
    position: DRAWER_POSITIONS.LEFT,
    transitionDuration: 250,
    type: DRAWER_TYPES.OVER
};
DrawerComponent.decorators = [
    { type: Component, args: [{
                selector: `${config.components.prefix}-drawer }`,
                template: "<div #drawer\n  [ngClass]=\"[prefix + '-drawer', className, type, position]\"\n  [class.open]=\"isOpen\"\n  [class.backdrop]=\"hasBackdrop\"\n>\n  <div [ngClass]=\"prefix + '-drawer-backdrop'\" #backdrop></div>\n\n  <div [ngClass]=\"prefix + '-drawer-content'\">\n    <ng-content select=\"materialize-drawer-content\"></ng-content>\n  </div>\n\n  <div [ngClass]=\"prefix + '-drawer-container'\">\n    <ng-content select=\"materialize-drawer-container\"></ng-content>\n  </div>\n</div>\n"
            }] }
];
/** @nocollapse */
DrawerComponent.ctorParameters = () => [
    { type: Renderer2 }
];
DrawerComponent.propDecorators = {
    backdropRef: [{ type: ViewChild, args: ['backdrop', { static: true },] }],
    drawerRef: [{ type: ViewChild, args: ['drawer', { static: true },] }],
    onOpenEmitter: [{ type: Output, args: ['onOpen',] }],
    onCloseEmitter: [{ type: Output, args: ['onClose',] }],
    className: [{ type: Input }],
    closeOnBackdrop: [{ type: Input }],
    hasBackdrop: [{ type: Input }],
    isOpen: [{ type: Input }],
    position: [{ type: Input }],
    transitionDuration: [{ type: Input }],
    type: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    DrawerComponent.defaultProps;
    /** @type {?} */
    DrawerComponent.prototype.backdropRef;
    /** @type {?} */
    DrawerComponent.prototype.drawerRef;
    /** @type {?} */
    DrawerComponent.prototype.onOpenEmitter;
    /** @type {?} */
    DrawerComponent.prototype.onCloseEmitter;
    /** @type {?} */
    DrawerComponent.prototype.className;
    /** @type {?} */
    DrawerComponent.prototype.closeOnBackdrop;
    /** @type {?} */
    DrawerComponent.prototype.hasBackdrop;
    /** @type {?} */
    DrawerComponent.prototype.isOpen;
    /** @type {?} */
    DrawerComponent.prototype.position;
    /** @type {?} */
    DrawerComponent.prototype.transitionDuration;
    /** @type {?} */
    DrawerComponent.prototype.type;
    /** @type {?} */
    DrawerComponent.prototype.prefix;
    /**
     * @type {?}
     * @private
     */
    DrawerComponent.prototype.renderer;
}
