// materialize-angular/esm2015/app/completed-components/accordion/accordion.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @license
 * Copyright Workylab. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://raw.githubusercontent.com/workylab/materialize-angular/master/LICENSE
 */
import { Component, ContentChildren, EventEmitter, Input, Output, QueryList } from '@angular/core';
import { CollapsibleComponent } from '../collapsible/collapsible.component';
import { config } from '../../config';
export class AccordionComponent {
    constructor() {
        this.className = AccordionComponent.defaultProps.className;
        this.onToggleEmitter = new EventEmitter();
        this.activeIndex = AccordionComponent.defaultProps.activeIndex;
    }
    /**
     * @return {?}
     */
    ngAfterContentInit() {
        this.collapsibles.forEach((/**
         * @param {?} item
         * @param {?} index
         * @return {?}
         */
        (item, index) => {
            item.onOpenEmitter.subscribe((/**
             * @return {?}
             */
            () => {
                this.toggleCollapsibles(index);
            }));
        }));
    }
    /**
     * @param {?} currentIndex
     * @return {?}
     */
    toggleCollapsibles(currentIndex) {
        this.activeIndex = currentIndex;
        this.onToggleEmitter.emit(this.activeIndex);
        this.collapsibles.forEach((/**
         * @param {?} item
         * @param {?} index
         * @return {?}
         */
        (item, index) => {
            if (index === currentIndex) {
                item.open();
            }
            else {
                item.close();
            }
        }));
    }
    /**
     * @return {?}
     */
    showNext() {
        /** @type {?} */
        const nextIndex = this.activeIndex + 1;
        this.toggleCollapsibles(nextIndex);
    }
    /**
     * @return {?}
     */
    showPrev() {
        /** @type {?} */
        const prevIndex = this.activeIndex - 1;
        this.toggleCollapsibles(prevIndex);
    }
}
AccordionComponent.defaultProps = {
    activeIndex: 0,
    className: ''
};
AccordionComponent.decorators = [
    { type: Component, args: [{
                selector: `${config.components.prefix}-accordion }`,
                template: "<div [ngClass]=\"className\">\n  <ng-content select=\"materialize-collapsible\"></ng-content>\n</div>\n"
            }] }
];
/** @nocollapse */
AccordionComponent.ctorParameters = () => [];
AccordionComponent.propDecorators = {
    collapsibles: [{ type: ContentChildren, args: [CollapsibleComponent,] }],
    onToggleEmitter: [{ type: Output, args: ['onToggle',] }],
    className: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    AccordionComponent.defaultProps;
    /** @type {?} */
    AccordionComponent.prototype.collapsibles;
    /** @type {?} */
    AccordionComponent.prototype.onToggleEmitter;
    /** @type {?} */
    AccordionComponent.prototype.className;
    /** @type {?} */
    AccordionComponent.prototype.activeIndex;
}
