// materialize-angular/esm2015/app/completed-components/tab-group/tab/tab.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @license
 * Copyright Workylab. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://raw.githubusercontent.com/workylab/materialize-angular/master/LICENSE
 */
import { Component, Input } from '@angular/core';
import { config } from '../../../config';
export class TabComponent {
    constructor() {
        this.className = TabComponent.defaultProps.className;
        this.disabled = TabComponent.defaultProps.disabled;
        this.link = TabComponent.defaultProps.link;
        this.title = TabComponent.defaultProps.title;
        this.isActive = false;
    }
}
TabComponent.defaultProps = {
    className: '',
    disabled: false,
    link: '',
    title: ''
};
TabComponent.decorators = [
    { type: Component, args: [{
                selector: `${config.components.prefix}-tab }`,
                template: "<div [hidden]=\"!isActive\" [ngClass]=\"className\">\n  <ng-content></ng-content>\n</div>\n"
            }] }
];
/** @nocollapse */
TabComponent.ctorParameters = () => [];
TabComponent.propDecorators = {
    className: [{ type: Input }],
    disabled: [{ type: Input }],
    link: [{ type: Input }],
    title: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    TabComponent.defaultProps;
    /** @type {?} */
    TabComponent.prototype.className;
    /** @type {?} */
    TabComponent.prototype.disabled;
    /** @type {?} */
    TabComponent.prototype.link;
    /** @type {?} */
    TabComponent.prototype.title;
    /** @type {?} */
    TabComponent.prototype.isActive;
}
